packages/host/app/components/ai-assistant/attached-file-dropdown-menu.gts

```diff
@@ -0,0 +1,176 @@
+import { on } from '@ember/modifier';
+import { action } from '@ember/object';
+import { service } from '@ember/service';
+import Component from '@glimmer/component';
+
+import { tracked } from '@glimmer/tracking';
+
+import Copy from '@cardstack/boxel-icons/copy';
+
+import Undo2 from '@cardstack/boxel-icons/undo-2';
+
+import { dropTask } from 'ember-concurrency';
+import perform from 'ember-concurrency/helpers/perform';
+import ToElsewhere from 'ember-elsewhere/components/to-elsewhere';
+
+import {
+  BoxelDropdown,
+  IconButton,
+  Menu,
+} from '@cardstack/boxel-ui/components';
+import { MenuItem } from '@cardstack/boxel-ui/helpers';
+
+import { ThreeDotsHorizontal, IconCode } from '@cardstack/boxel-ui/icons';
+
+import RestorePatchedFileModal from '@cardstack/host/components/ai-assistant/restore-file-modal';
+import CardService from '@cardstack/host/services/card-service';
+import MatrixService from '@cardstack/host/services/matrix-service';
+import OperatorModeStateService from '@cardstack/host/services/operator-mode-state-service';
+
+import { type FileDef } from 'https://cardstack.com/base/file-api';
+
+import { Submodes } from '../submode-switcher';
+
+export default class AttachedFileDropdownMenu extends Component<{
+  Args: {
+    file: FileDef;
+    isNewFile: boolean;
+    version?: 'diff-editor';
+  };
+}> {
+  @service declare operatorModeStateService: OperatorModeStateService;
+  @service declare matrixService: MatrixService;
+  @service declare cardService: CardService;
+  @tracked isRestorePatchedFileModalOpen = false;
+  @tracked fileContent: string | null = null;
+  @action private openInCodeMode() {
+    if (
+      this.operatorModeStateService.state.submode === Submodes.Code &&
+      this.operatorModeStateService.state.codePath?.toString() ===
+        this.args.file.sourceUrl
+    ) {
+      return;
+    }
+    this.operatorModeStateService.updateSubmode(Submodes.Code);
+    this.operatorModeStateService.updateCodePath(
+      new URL(this.args.file.sourceUrl!),
+    );
+  }
+
+  copySubmittedContent = async () => {
```

Comment:
this casts a EC task to a promise which bascially removes the cancellability and will throw if cancelled. Please only use EC tasks in a component.